pkg/scalers/predictkube_scaler.go

```diff
@@ -0,0 +1,463 @@
+				for _, v := range val.Values {
+					t, err := tc.AdaptTimeToPbTimestamp(tc.TimeToTimePtr(v.Timestamp.Time()))
+					if err != nil {
+						return nil, err
+					}
+
+					out = append(out, &commonproto.Item{
+						Timestamp:  t,
+						Value:      float64(v.Value),
+						MetricName: metricName,
+					})
+				}
+			}
+		}
+	case model.ValScalar:
+		if res, ok := result.(*model.Scalar); ok {
+			t, err := tc.AdaptTimeToPbTimestamp(tc.TimeToTimePtr(res.Timestamp.Time()))
+			if err != nil {
+				return nil, err
+			}
+
+			out = append(out, &commonproto.Item{
+				Timestamp:  t,
+				Value:      float64(res.Value),
+				MetricName: metricName,
+			})
+		}
+	case model.ValString:
+		if res, ok := result.(*model.String); ok {
+			t, err := tc.AdaptTimeToPbTimestamp(tc.TimeToTimePtr(res.Timestamp.Time()))
+			if err != nil {
+				return nil, err
+			}
+
+			s, err := strconv.ParseFloat(res.Value, 64)
+			if err != nil {
+				return nil, err
+			}
+
+			out = append(out, &commonproto.Item{
+				Timestamp:  t,
+				Value:      s,
+				MetricName: metricName,
+			})
+		}
+	}
+
+	return out, nil
+}
+
+func parsePredictKubeMetadata(config *ScalerConfig) (result *predictKubeMetadata, err error) {
+	validate := validator.New()
+	meta := predictKubeMetadata{}
+
+	if val, ok := config.TriggerMetadata["query"]; ok {
+		if len(val) == 0 {
+			return nil, fmt.Errorf("no query given")
+		}
+
+		meta.query = val
+	} else {
+		return nil, fmt.Errorf("no query given")
+	}
+
+	if val, ok := config.TriggerMetadata["prometheusAddress"]; ok {
+		err = validate.Var(val, "url")
+		if err != nil {
+			return nil, fmt.Errorf("invalid prometheusAddress")
+		}
+
+		meta.prometheusAddress = val
+	} else {
+		return nil, fmt.Errorf("no prometheusAddress given")
+	}
+
+	if val, ok := config.TriggerMetadata["predictHorizon"]; ok {
+		meta.predictHorizon, err = str2duration.ParseDuration(val)
+		if err != nil {
+			return nil, fmt.Errorf("predictHorizon parsing error %s", err.Error())
+		}
+	} else {
+		return nil, fmt.Errorf("no predictHorizon given")
+	}
+
+	if val, ok := config.TriggerMetadata["queryStep"]; ok {
+		meta.stepDuration, err = str2duration.ParseDuration(val)
+		if err != nil {
+			return nil, fmt.Errorf("queryStep parsing error %s", err.Error())
+		}
+	} else {
+		return nil, fmt.Errorf("no queryStep given")
+	}
+
+	if val, ok := config.TriggerMetadata["historyTimeWindow"]; ok {
+		meta.historyTimeWindow, err = str2duration.ParseDuration(val)
+		if err != nil {
+			return nil, fmt.Errorf("historyTimeWindow parsing error %s", err.Error())
+		}
+	} else {
+		return nil, fmt.Errorf("no historyTimeWindow given")
+	}
+
+	if val, ok := config.TriggerMetadata["threshold"]; ok {
+		meta.threshold, err = strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("threshold parsing error %s", err.Error())
+		}
+	} else {
+		return nil, fmt.Errorf("no threshold given")
+	}
+
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
```

Comment:
Is the metric name necessary during the process or it is only for UX experience? I mean, we can ensure that the metric name is unique because we append sX so if it's possible, I prefer not giving the option to set it in order to make it easiest as possible but maybe is in use for other stuffs

Reply:
+1

We have recently reimplemented the way how metricNames are being generated, so if it is not needed we shouldn't bother users with specifying this in the property.

Reply:
done

Reply:
+1